agent/api/containerstatus.go

```diff
@@ -49,26 +57,50 @@ func (cs ContainerStatus) String() string {
 	return "NONE"
 }
 
-// TaskStatus maps the container status to the corresponding task status
-func (cs *ContainerStatus) TaskStatus() TaskStatus {
+// TaskStatus maps the container status to the corresponding task status. The
+// transition map is illustrated below.
+//
+// Container: None -> Pulled -> Created -> Running -> Provisioned -> Stopped -> Zombie
+//
+// Task     : None ->     Created       ->         Running        -> Stopped
+func (cs *ContainerStatus) TaskStatus(steadyStateStatus ContainerStatus) TaskStatus {
 	switch *cs {
 	case ContainerStatusNone:
 		return TaskStatusNone
+	case steadyStateStatus:
+		return TaskRunning
 	case ContainerCreated:
 		return TaskCreated
-	case ContainerRunning:
-		return TaskRunning
 	case ContainerStopped:
 		return TaskStopped
 	}
+
+	if *cs == ContainerRunning && steadyStateStatus == ContainerResourcesProvisioned {
```

Comment:
Although it can work, it seems like adding potentially duplicate `case`s in `switch` and might seem unintuitive. The current code sort of makes it explicit as to why and what the special case is being handled here. Let me know if you think there's a better way to do this. Using `map` might also be an option here.